File: Project4/For_Students/matlab/exact_solution.m
%---------------------------------------------------------------------%
%This function computes the Initial and Analytic Solutions.
%Written by F.X. Giraldo on 10/2003
%           Department of Applied Mathematics
%           Naval Postgraduate School 
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function [qe,fe] = exact_solution(coord,npoin,c)

%Initialize
qe=zeros(npoin,1);
fe=zeros(npoin,1);
cc=pi*c;

%Generate Grid Points
for i=1:npoin
    x=coord(1,i);
    y=coord(2,i);
    qe(i)=sin(cc*x)*sin(cc*y);
    fe(i)=-2*cc^2*sin(cc*x)*sin(cc*y);
end %ip      


      

File: Projects/Project_07_3D_Elliptic/For_Students/julia/exact_solution.jl
#---------------------------------------------------------------------#
#This function computes the Initial & Analytic Solutions.
#Written by F.X. Giraldo on 7/2021
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function exact_solution(coord,Npoin,c,DFloat)

    #Initialize
    qe=zeros(DFloat,Npoin,1)
    fe=zeros(DFloat,Npoin,1)
    cc=c*π

    #Generate Grid Points
    for I=1:Npoin
        x=coord[1,I]
        y=coord[2,I]
        z=coord[3,I]
        qe[I]=sin(cc*x)*sin(cc*y)*sin(cc*z)
        fe[I]=-3*cc^2*sin(cc*x)*sin(cc*y)*sin(cc*z)
    end #I

    return (qe,fe)

end


